airbyte-integrations/integration-test-lib/src/main/java/io/airbyte/integrations/base/TestDestination.java

```diff
@@ -0,0 +1,243 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.base;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Lists;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.lang.Exceptions;
+import io.airbyte.commons.resources.MoreResources;
+import io.airbyte.config.DestinationConnectionImplementation;
+import io.airbyte.config.Schema;
+import io.airbyte.config.StandardSync;
+import io.airbyte.config.StandardTargetConfig;
+import io.airbyte.singer.SingerMessage;
+import io.airbyte.singer.SingerMessage.Type;
+import io.airbyte.workers.WorkerException;
+import io.airbyte.workers.process.DockerProcessBuilderFactory;
+import io.airbyte.workers.process.ProcessBuilderFactory;
+import io.airbyte.workers.protocols.singer.DefaultSingerTarget;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+public abstract class TestDestination {
+
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  private TestDestinationEnv testEnv;
+
+  private Path jobRoot;
+  protected Path localRoot;
+  private ProcessBuilderFactory pbf;
+
+  /**
+   * Name of the docker image that the tests will run against.
+   *
+   * @return docker image name
+   */
+  protected abstract String getImageName();
+
+  /**
+   * Configuration specific to the integration. Will be passed to integration where appropriate in
+   * each test. Should be valid.
+   *
+   * @return integration-specific configuration
+   */
+  protected abstract JsonNode getConfig();
+
+  /**
+   * Configuration specific to the integration. Will be passed to integration where appropriate in
+   * tests that test behavior when configuration is invalid. e.g incorrect password. Should be
+   * invalid.
+   *
+   * @return integration-specific configuration
+   */
+  protected abstract JsonNode getInvalidConfig();
```

Comment:
Invalid is a bit misleading. Is it bad format or that it will fail the check?

Reply:
this one: "that it will fail the check". any suggestions on names? i agree that current name is not clear enough. maybe just FailingConfig or something.

Reply:
`getFailCheckConfig`?